src/openforms/forms/fd_translations_converter.py

```diff
@@ -0,0 +1,176 @@
+"""
+Implements the translation converter for form definitions.
+
+Github issue: https://github.com/open-formulieren/open-forms/issues/2958
+
+Before, translations for formio components were saved in the JSONField
+``FormDefinition.component_translations``, which is a key-value mapping of language
+code to translations mapping, which itself is a key-value mapping of source string to
+translation string. This had the drawback that if different components happen to use the
+same literal, they also shared the same translation.
```

Comment:
So this means that if two components were defined from a reusable form definition, they couldn't have custom translations?

Reply:
Couldn't and can't - as discussed on Slack, a FD is reusable as a whole or not at all. Translations are baked in and not contextualized in a form - which is as intended.